langserver/references.go

```diff
@@ -296,6 +296,10 @@ func refStreamAndCollect(ctx context.Context, conn jsonrpc2.JSONRPC2, req *jsonr
 				send()
 				return locs
 			}
+			if len(locs) >= limit {
+				stop()
```

Comment:
just to make sure I read this correct, this limit parameter is just best effort. There may be more than limit number of results returned?

Is there a reason you chose this over limiting the results on the producer side of the refs channel?